datafusion-cli/src/catalog.rs

```diff
@@ -199,6 +201,16 @@ impl SchemaProvider for DynamicFileSchemaProvider {
         self.inner.table_exist(name)
     }
 }
+fn substitute_tilde(cur: String) -> String {
+    if let Some(usr_dir_path) = home_dir() {
+        if let Some(usr_dir) = usr_dir_path.to_str() {
+            if cur.starts_with('~') && !usr_dir.is_empty() {
```

Comment:
Should we support this on Windows as well? From this [post](https://superuser.com/q/332871/1597536), this sign is only supported by Powershell

Reply:
I don't have a strong preference -- making datafusion-cli more consistent with windows norms seems like a good idea to me but I am not a (current user).